src/classes/tmControllerTrialFeatureRequest.cls

```diff
@@ -0,0 +1,486 @@
+            }
+            return countryOptions;
+        }
+        private set;
+    }
+
+    public Boolean isCountryEditable {
+        get {
+            return tfrDomain.isDemoAccount && !tfrDomain.hasValidIsoCountryCode;
+        }
+        private set;
+    }
+
+    public String selectedCountryName {
+        get {
+            for (sfbase__CountryRevenueMap__c countryMap : countryList) {
+                if (tfrDomain.countryCode.equalsIgnoreCase(countryMap.sfbase__CountryISOCode__c)) {
+                    selectedCountryName  = countryMap.sfbase__CountryName__c;
+                    break;
+                }
+            }
+            return selectedCountryName;
+        }
+        private set;        
+    }
+
+    private String objectIdForLog {
+        get {
+
+            return tfrDomain!= null && tfrDomain.tfrDecorator != null && tfrDomain.tfrDecorator.tfr !=null && tfrDomain.tfrDecorator.tfr.Id != null ?  tfrDomain.tfrDecorator.tfr.Id  : tfrAccountId ;
+        }
+        private set;        
+    }
+
+    /*
+    *   Save the TFR and if it is an insert then take the user to the Edit page for that record. If
+    *       The TFR already exists then only refresh page with a return null.
+    */
+    public PageReference saveTfr() {
+        showSaved = true;
+        try {
+            updateUpgradeInfoToTFR();
+            tfrDomain.selectedProducts = orgProducts;
+            if (tfrDomain.tfrDecorator.canEdit) {
+                tfrDomain.save();
+            }
+            
+            if (String.isBlank(tfrId)) {
+                tmTrialFeatureRequest__c t = new tmTrialFeatureRequest__c(Id = tfrDomain.tfrDecorator.tfr.Id);
+                ApexPages.StandardController sc = new ApexPages.StandardController(t);
+                PageReference p = sc.edit();
+                p.getParameters().put('id', tfrDomain.tfrDecorator.tfr.Id);
+                p.getParameters().put('showSaveMsg', 'True');
+                return p;
+            } else {
+                return null;
+            }
+        } catch (tmException tmEx) {
+            showSaved = false;
+            for (String s: tmEx.errors) {
+                ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.FATAL,s));
+            }
+            tmOrgLog.log( BaseOrgLog.level.INFO,'tmControllerTrialFeatureRequest.saveTfr',tmEx.errorString,objectIdForLog);
+            return null;            
+        } catch (Exception ex) {
+            showSaved = false;
+            showForm = false; 
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.FATAL,tmConstants.ERROR_SAVING_TRIAL_FEATURE_REQUEST));
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.FATAL,ex.getMessage()));
+            tmOrgLog.log( BaseOrgLog.level.ERROR,'tmControllerTrialFeatureRequest.saveTfr',ex.getMessage(),objectIdForLog);
+
+            return null;
+        }           
+    }
+
+    private void updateUpgradeInfoToTFR() {
```

Comment:
Methods must use the `lowerCamelCase` naming convention. Please rename accordingly.